include/seqan3/alignment/pairwise/policy/find_optimum_policy.hpp

```diff
@@ -162,18 +153,30 @@ class find_optimum_policy
             derived_t const * me = static_cast<derived_t const *>(this);
             std::ranges::for_each(views::zip(*me->score_matrix_iter, *me->trace_matrix_iter), [&](auto && cell)
             {
-                cache.optimum = (get<0>(cell).current > cache.optimum.score)
-                                ? alignment_optimum{get<0>(cell).current, get<1>(cell).coordinate}
-                                : cache.optimum;
+                check_and_update(cell, cache);
             });
         }
         else  // Only check the last cell for the global alignment.
         {
-            cache.optimum = (get<0>(current_cell).current > cache.optimum.score)
-                                ? alignment_optimum{get<0>(current_cell).current, get<1>(current_cell).coordinate}
-                                : cache.optimum;
+            check_and_update(current_cell, cache);
         }
     }
+
+    /*!\brief Tests if the score in the current cell is greater than the current alignment optimum.
+     * \tparam cell_t The type of the alignment matrix cell.
+     * \tparam score_t The alignment algorithm score type.
+     *
+     * \param[in] cell The current cell to get the score and the coordinate from.
+     * \param[in,out] state The state with the current optimum to update.
+     */
+    template <typename cell_t, typename score_t>
+    constexpr void check_and_update(cell_t const & cell, alignment_algorithm_cache<score_t> & state) const noexcept
```

Comment:
```suggestion
    constexpr void check_and_update(cell_t const & cell, alignment_algorithm_state<score_t> & state) const noexcept
```
Don't forget this after rebase.